src/components/chart/chart.tsx

```diff
@@ -0,0 +1,381 @@
+import { Component, h, Prop, Watch } from '@stencil/core';
+import { Languages } from '../date-picker/date.types';
+import translate from '../../global/translations';
+import { createRandomString } from '../../util/random-string';
+import { ChartItem } from './chart.types';
+
+const PERCENT = 100;
+const DEFAULT_AXIS_INCREMENT = 10;
+
+/**
+ * A chart is a graphical representation of data, in which
+ * visual symbols such as such bars, dots, lines, or slices, represent
+ * each data point, in comparison to others.
+ *
+ * @exampleComponent limel-example-chart-stacked-bar
+ * @exampleComponent limel-example-chart-orientation
+ * @exampleComponent limel-example-chart-max-value
+ * @exampleComponent limel-example-chart-type-bar
+ * @exampleComponent limel-example-chart-type-scatter
+ * @exampleComponent limel-example-chart-type-area
+ * @exampleComponent limel-example-chart-type-line
+ * @exampleComponent limel-example-chart-type-pie
+ * @exampleComponent limel-example-chart-type-doughnut
+ * @exampleComponent limel-example-chart-type-ring
+ * @exampleComponent limel-example-chart-type-gantt
+ * @exampleComponent limel-example-chart-multi-axis
+ * @exampleComponent limel-example-chart-multi-axis-with-negative-start-values
+ * @exampleComponent limel-example-chart-multi-axis-area-with-negative-start-values
+ * @exampleComponent limel-example-chart-axis-increment
+ * @exampleComponent limel-example-chart-accessibility
+ * @exampleComponent limel-example-chart-styling
+ * @exampleComponent limel-example-chart-creative-styling
+ * @beta
+ */
+
+@Component({
+    tag: 'limel-chart',
+    shadow: true,
+    styleUrl: 'chart.scss',
+})
+export class Chart {
+    /**
+     * Defines the language for translations.
+     * Will translate the translatable strings on the components.
+     */
+    @Prop({ reflect: true })
+    public language: Languages = 'en';
+
+    /**
+     * Helps users of assistive technologies to understand
+     * the context of the chart, and what is being displayed.
+     */
+    @Prop({ reflect: true })
+    public accessibleLabel?: string;
+
+    /**
+     * Helps users of assistive technologies to understand
+     * what the items in the chart represent.
+     */
+    @Prop({ reflect: true })
+    public accessibleItemsLabel?: string;
+
+    /**
+     * List of items in the chart,
+     * each representing a data point.
+     */
+    @Prop()
+    public items!: ChartItem[];
+
+    /**
+     * Defines how items are visualized in the chart.
+     */
+    @Prop({ reflect: true })
+    public type?:
+        | 'area'
+        | 'bar'
+        | 'doughnut'
+        | 'line'
+        | 'pie'
+        | 'ring'
+        | 'scatter'
+        | 'stacked-bar' = 'stacked-bar';
```

Comment:
When I hear "stacked bar" I think of something like this
![image](https://github.com/user-attachments/assets/4f49f07b-e494-4793-adff-269097c0c082)

Our example here only has one bar though so it seems more like a pie chart that has been flattened :smile: Perhaps we should call it something else unless we add support for multiple stacked bars

Reply:
I'd say these are chart-sets and require sending multiple arrays of items into the component. We should either make another component and call it for instance `chart-set`, or move the current one into a private sub-component. I don't know. 

Currently the consumer can make these chart-sets on their own, knowing a little CSS and having multiple `limel-charts`. But if we wanna handle them, we can of course. We (or the consumer) should just make sure that the min and max values in the datasets are the same, and the data has the same range. This means we need to add the `minValue` prop too, otherwise for those charts with X and Y axis, we will get the zero-line in different locations.

Also, such chart-sets (e.g. area or lines or bars) will be overlayed on each other. It means we will absolutely position each chart on the previous chart. There is no way in normal CSS and HTML to hover on the layers which are placed below the top-most layer. (we already have this issue with the pie chart, you can't hover the items since they are all layers on top of each other). This will be a downside of having such charts. We need to solve it then by proviing an interactive "Legend" where items inside the chart can be highlighted if you hover on their equivalents inside the legend.